### DISCUSSION TITLE
Mandarine v1.3.0 is coming, and you should be aware of this...

### DISCUSSION THREAD

**Discussion by andreespirela** [nestdotland/nest.land#246] [Show and tell]
Hello there!. I'm Andres Pirela, Author of [Mandarine.TS](http://mandarinets.org/).

<img src="https://www.mandarinets.org/assets/images/full-logo-simple.svg" width="180" height="180" />  

As of right now, we find ourselves working for version **1.3.0** which will be a major version with a lot of useful things coming.

- Same routes with different methods:
    - Before, Mandarine used didn't support same routes, you would get an error if you had the same route (even with different methods such as POST, GET, DELETE...). Now, you can have the same route with different methods.
- Wildcard routes
    - Mandarine now supports **wildcard routes** ex: `/(.*)` .
        - If a route **that is not wildcard** exist and a wildcard route exists that matches other routes, the routes that are not wildcard will be prioritized.
- We are also adding `greaterThan` and `lessThan` to MQL (Mandarine Query Language).

**Mandarine query language**. Many of you may not know what this is, and it is hard to explain. Basically, MQL transform the name of your methods into SQL queries.

```typescript
@Repository()
export abstract class MyRepository {
          public findByNameAndCountryIsNotAndAgeIsGreaterThan(name: string, country: string, age: number) {}
}
```

With Mandarine, you would inject `MyRepository`, and then when you call `findByNameAndCountryIsNotAndAgeIsGreaterThan` you would also be executing a database query according to your configured data source (as of right now, only postgres is supported)

```typescript
@Service()
export class MyComponent {
            constructor(myRepo: MyRepository) {}

            public static myHandler() {
                this.findByNameAndCountryAndAgeIsGreaterThan("Andres", "United States", 19);
            } 
}
```
In `myHandler` when calling `findByNameAndCountryIsNotAndAgeIsGreaterThan` that would be translated to the following DB query:
```sql
SELECT * FROM ....... WHERE name = 'Andres' AND country = 'United States' AND age > 19
```
The whole process is managed automatically by mandarine, and when calling your empty function `findByNameAndCountryAndAgeIsGreaterThan` you will be actually getting data from the database. This is all done by Mandarine.

For more information about [**MQL click here**](https://www.mandarinets.org/docs/master/mandarine/mandarine-query-language) 

-----------

Hint: Mandarine v2.0.0 planning has already started, and it's really nice!.

**Comment by andreespirela** [comment]
Like I always mention, Mandarine is currently looking for collaborators. if you feel you are that person, feel free to reach out to me or open an issue in our main repository.

Thanks NestLand for having this discussion tab and making it available for this type of presentations.

**Reply by t8** [reply]
Of course! Amazing to see where Mandarine is going. It's really interesting to see the addition of wildcard routes and your updates to MQL. 

Feel free to drop a link to your repository so that people interested can check out the code!
